spatialdata/_dl/datasets.py

```diff
@@ -0,0 +1,130 @@
+from typing import Callable, Optional
+
+import numpy as np
+from geopandas import GeoDataFrame
+from multiscale_spatial_image import MultiscaleSpatialImage
+from spatial_image import SpatialImage
+from torch.utils.data import Dataset
+
+from spatialdata import SpatialData
+from spatialdata._core._rasterize import rasterize
+from spatialdata._core._spatialdata_ops import get_transformation
+from spatialdata._core.core_utils import get_dims
+from spatialdata._core.models import (
+    Image2DModel,
+    Image3DModel,
+    Labels2DModel,
+    Labels3DModel,
+    ShapesModel,
+    get_schema,
+)
+from spatialdata._core.transformations import BaseTransformation
+from spatialdata.utils import affine_matrix_multiplication
+
+
+class ImageTilesDataset(Dataset):
+    def __init__(
+        self,
+        sdata: SpatialData,
+        regions_to_images: dict[str, str],
+        tile_dim_in_units: float,
+        tile_dim_in_pixels: int,
+        target_coordinate_system: str = "global",
+        transform: Optional[Callable[[SpatialData], dict[str, SpatialImage]]] = None,
```

Comment:
please add a docstring. I think the input parameters aren't clear (e.g., `tile_dim_in_units` vs. `tile_dim_in_pixels`)